src/AiAdapter.js

```diff
@@ -496,15 +488,15 @@ var ticTacToe = ticTacToe || {};
    * Determines whether a square is empty.
    * @param {integer} square
    */
-  app.AiAdapter.prototype.squareIsEmpty = function (square) {
-    return (square === 0) ? true : false;
+  app.AiAdapter.prototype.isSquareEmpty = function (square, board) {
```

Comment:
the player needs to know whether a square is empty, though. do you suggest also passing the Game object to the Player so it could do Game.isSquareEmpty(square); ? It is still the same feature envy problem.